packages/salesforcedx-vscode-core/src/commands/forceSourceDeploy.ts

```diff
@@ -7,60 +7,34 @@
 
 import {
   CliCommandExecutor,
-  Command,
-  ForceDeployErrorParser,
-  SfdxCommandBuilder
+  ForceDeployErrorParser
 } from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
 import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
 import * as vscode from 'vscode';
 import { handleDiagnosticErrors } from '../diagnostics';
-import { nls } from '../messages';
 import { telemetryService } from '../telemetry';
-import {
-  SfdxCommandlet,
-  SfdxCommandletExecutor,
-  SfdxWorkspaceChecker
-} from './commands';
-import {
-  FileType,
-  ManifestOrSourcePathGatherer,
-  SelectedPath
-} from './forceSourceRetrieve';
+import { SfdxCommandletExecutor } from './commands';
 
 vscode.workspace.onDidChangeTextDocument(e => {
   if (ForceSourceDeployExecutor.errorCollection.has(e.document.uri)) {
     ForceSourceDeployExecutor.errorCollection.delete(e.document.uri);
   }
 });
 
-export class ForceSourceDeployExecutor extends SfdxCommandletExecutor<
-  SelectedPath
+export abstract class ForceSourceDeployExecutor extends SfdxCommandletExecutor<
```

Comment:
Awesome, this makes every type of deploy command render any deploy errors in the problem view.